### PR TITLE
The Witness: The big dumb refactor

### PR DESCRIPTION
**Pull request by NewSoupVi**
The structure of The Witness is some actual hot trash. Also it violates style a lot. Also it uses old APIs.

- Ran `ruff --fix`
- Made "static classes" into modules because Jesus Fcking Christ
- Changed some small things here and there
- Got rid of dumb naming like `regio` and `locat`
- Cried at the thought of all the conflicts I will have to fix soon

### CODE REVIEW

**Comment by NewSoupVi** [comment]
For any potential reviewers: There is 0 new code here. Everything just moved around. If you see something that looks new, it's removed from a different file.

**Comment by NewSoupVi** [comment]
Included the `ruff.toml` that was used for this.

Made sure it actually passes, except for one thing that will be addressed by this PR: https://github.com/ArchipelagoMW/Archipelago/pull/3049

File: worlds/witness/data/static_logic.py [hunk 1]
```diff
@@ -199,102 +169,95 @@ def __init__(self, lines=None):
...
-    @lazy
-    def sigma_expert(self) -> StaticWitnessLogicObj:
-        return StaticWitnessLogicObj(get_sigma_expert_logic())
 
-    @lazy
-    def sigma_normal(self) -> StaticWitnessLogicObj:
-        return StaticWitnessLogicObj(get_sigma_normal_logic())
+@lru_cache
+def get_sigma_expert() -> StaticWitnessLogicObj:
+    return StaticWitnessLogicObj(get_vanilla_logic())
```
Comment (Exempt-Medic) [inline worlds/witness/data/static_logic.py:236]: Are vanilla and expert flipped?

Comment (NewSoupVi) [reply worlds/witness/data/static_logic.py:236]: LMAO

Comment (NewSoupVi) [reply worlds/witness/data/static_logic.py]: Never trust me near code. Never

**Review by Exempt-Medic — APPROVED** [review]
Carefully read through all the changes, merged into `main` and generated five hundred 10-player games and one thousand single- and two-player games. Compared 0.4.5 and refactor spoiler logs (they were same except for playthrough) and walked through the playthrough checking for errors.

File: worlds/witness/__init__.py [hunk 2]
```diff
@@ -129,18 +128,18 @@ def determine_sufficient_progression(self):
                             f" progression items that can be placed in other players' worlds. Please turn on Symbol"
                             f" Shuffle, Door Shuffle or Obelisk Keys.")
```
Comment (ScipioWright) [inline worlds/witness/__init__.py:129]:
```suggestion
                            f" Shuffle, Door Shuffle, or Obelisk Keys.")
```
This was close enough to changes that I could suggest it. Oxford comma my beloved

**Review by ScipioWright — APPROVED** [review]
Read through changes, they sure are changes. Did a bunch of test gens with 50 player worlds with some triggers to avoid that exception regarding low progression counts.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**